api/src/main/java/org/bahmni/module/bahmni/ie/apps/service/impl/BahmniFormPrivilegesServiceImpl.java

```diff
@@ -0,0 +1,151 @@
+package org.bahmni.module.bahmni.ie.apps.service.impl;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.bahmni.module.bahmni.ie.apps.dao.BahmniFormPrivilegeDao;
+import org.bahmni.module.bahmni.ie.apps.model.FormPrivilege;
+import org.bahmni.module.bahmni.ie.apps.service.BahmniFormPrivilegesService;
+
+import org.openmrs.api.APIException;
+
+import org.openmrs.api.impl.BaseOpenmrsService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.*;
+
+@Service("bahmniFormPrivilegesService")
+public class BahmniFormPrivilegesServiceImpl extends BaseOpenmrsService implements BahmniFormPrivilegesService {
+
+    private BahmniFormPrivilegeDao bahmniFormPrivilegeDao;
+    protected Log log = LogFactory.getLog(getClass());
+    private BahmniFormPrivilegesService bahmniFormPrivilegesService;
+
+    @Autowired
+    public BahmniFormPrivilegesServiceImpl(BahmniFormPrivilegeDao bahmniFormPrivilegeDao) {
+        this.bahmniFormPrivilegeDao = bahmniFormPrivilegeDao;
+        this.bahmniFormPrivilegesService = bahmniFormPrivilegesService;
+    }
+    public BahmniFormPrivilegesServiceImpl(){
+
+    }
+
+    public FormPrivilege saveFormPrivilege(FormPrivilege toPersistFormPrivilege) throws APIException {
+
+        if (toPersistFormPrivilege != null) {
+            FormPrivilege originalFormPrivilege ;
+            originalFormPrivilege = getFormPrivilege(toPersistFormPrivilege.getPrivilegeName(),toPersistFormPrivilege.getFormId());
+
+            if (originalFormPrivilege != null) {
+                originalFormPrivilege.setFormId(toPersistFormPrivilege.getFormId());
+                originalFormPrivilege.setPrivilegeName(toPersistFormPrivilege.getPrivilegeName());
+                originalFormPrivilege.setEditable(toPersistFormPrivilege.getEditable());
+                originalFormPrivilege.setViewable(toPersistFormPrivilege.getViewable());
+                toPersistFormPrivilege = originalFormPrivilege;
+            }
+
+            return bahmniFormPrivilegeDao.saveFormPrivilege(toPersistFormPrivilege);
+        } else {
+            return null;
+        }
+    }
+
+    @Override
+    @Transactional
+    public List<FormPrivilege> saveFormPrivileges(List<FormPrivilege> formPrivileges) {
+        log.info("Inside BahmniFormPrivilegesServiceImpl -->saveFormPrivileges " + formPrivileges);
+        List<FormPrivilege> resultList = new ArrayList<FormPrivilege>();
+        List<FormPrivilege> privilegesList = new ArrayList<FormPrivilege>();
+        List<FormPrivilege> oldPrivilegeList = new ArrayList<FormPrivilege>();
+
+//        if (formPrivileges.size() == 1 && formPrivileges.get(0).getPrivilegeName().equalsIgnoreCase("")) {
+//            oldPrivilegeList = getAllPrivilegesForForm(formPrivileges.get(1).getFormId());
```

Comment:
@buvaneswari-arun is this commented code is required to be here ?

Reply:
Removed